providers/oidc.go

```diff
@@ -181,6 +184,7 @@ func (p *OIDCProvider) CreateSessionStateFromBearerToken(ctx context.Context, ra
 	newSession.IDToken = rawIDToken
 	newSession.RefreshToken = ""
 	newSession.ExpiresOn = &idToken.Expiry
+	newSession.Groups = strings.Join(p.RequiredGroups, ",")
```

Comment:
I think this should happen in `createSessionStateInternal` -- and when it happens there, I think we want the session.Groups to be what we extract from the `groups` claim, not from the `RequiredGroups` option.

(I think we want to pass the `groups` claim in full to the backend in the X-headers and in the session in case the backend wants to do alternative  authz instead of `--required-groups`)

Reply:
I thought about this, and saw some other solutions like https://github.com/buzzfeed/sso/blob/master/docs/sso_config.md#headers passing only the "required" groups hence going that direction, can swap over though without much fuss.

Reply:
Ok change made, and removed the duplicate setting, I already had it in `createSessionStateInternal` and the duplication was an artifact of some testing that isn't actualy needed.

Reply:
> Ok change made, and removed the duplicate setting, I already had it in `createSessionStateInternal` and the duplication was an artifact of some testing that isn't actualy needed.

I'm waiting for the other maintainers to weigh in, I don't know if we do any authz anywhere in the codebase already -- or if we purely do authn and delegate authz to backends that they can figure out via headers.

Hence I'm much more interested in getting the X-Forwarded-Groups header passing robust and accurate to the IDToken claims and we can work out the authz `--required-groups` aspect of this PR as well if we want some minor authz logic in the codebase.

Reply:
No problems thanks for taking a look! from my perspective our main need is to have the authz part handled in the proxy to simplify our deployments so it would be great if it could stay.